src/ViewerOptions.tsx

```diff
@@ -0,0 +1,199 @@
+            options: { checkExisting: true, label: formElements.name.value },
+            representation: formElements.style.value,
+        });
+    }
+
+    async function colour(event: React.SyntheticEvent<HTMLFormElement>) {
+        event.preventDefault();
+        const form = event.currentTarget;
+        const formElements = form.elements as typeof form.elements & {
+            start: { value: number };
+            end: { value: number };
+            colourPicker: { value: string };
+        };
+        const range = Array.from(
+            { length: formElements.end.value - formElements.start.value },
+            (v, k) => k + Number(formElements.start.value)
+        );
+
+        const data =
+            plugin.current!.managers.structure.hierarchy.current.structures[0]
+                ?.cell.obj?.data;
+        if (!data) return;
+
+        const selection = Script.getStructureSelection(
+            (Q) =>
+                Q.struct.generator.atomGroups({
+                    "residue-test": Q.core.set.has([
+                        Q.set(...range),
+                        Q.ammp("auth_seq_id"),
+                    ]),
+                    "group-by":
+                        Q.struct.atomProperty.macromolecular.residueKey(),
+                }),
+            data
+        );
+        const lociGetter = async (s: Structure) =>
+            StructureSelection.toLociWithSourceUnits(selection);
+
+        for (const s of plugin.current!.managers.structure.hierarchy.current
+            .structures) {
+            const components = s.components;
+            setStructureOverpaint(
+                plugin.current!,
+                components,
+                Color(
+                    Number(`0x${formElements.colourPicker.value.substring(1)}`)
+                ),
+                lociGetter
+            );
+        }
+    }
+
+    const inputStyle = {
+        width: 200,
+        // marginBottom: 10,
+        padding: 2,
+    };
+
+    const labelStyle = {
+        display: "block",
+        width: 50,
+    };
+
+    return (
+        <>
+            <h5>Create Component</h5>
+            <form onSubmit={createComponent}>
+                <label style={labelStyle}>
+                    Start
+                    <input type="number" id="start" style={inputStyle} />
+                </label>
+                <br />
+                <label style={labelStyle}>
+                    End
+                    <input type="number" id="end" style={inputStyle} />
+                </label>
+                <br />
+                <label style={labelStyle}>
+                    Name
+                    <input type="text" id="name" style={inputStyle} />
+                </label>
+                <br />
+                <label style={labelStyle}>
+                    Style
+                    <select id="style" style={inputStyle}>
+                        <option value="cartoon">Cartoon</option>
```

Comment:
Are these values from anywhere in particular, is it worth defining it as a constant somewhere, eg:

```typescript
const STYLE_OPTIONS = {
    "backbone": "Backbone",
    // ....
}

// ....

Object.entries([key, label] => <option key={key} value={key}>{label}</option>
```

Reply:
Added constant with source